src/univariate/discrete/geometric.jl

```diff
@@ -67,6 +67,14 @@ kurtosis(d::Geometric) = 6 + abs2(d.p) / (1 - d.p)
 
 entropy(d::Geometric) = (-xlogx(succprob(d)) - xlogx(failprob(d))) / d.p
 
+function kldivergence(p::Geometric, q::Geometric)
+    x = succprob(p)
+    y = succprob(q)
+    z = failprob(p)
+    w = failprob(q)
+    return log(x / y) + z / x * log(z / w)
```

Comment:
I think you want to use
```suggestion
    return log(x / y) + xlogy(z / x, z / w)
```
to avoid NaN values in the case `z = 0`. Maybe more changes or even branches are needed for other special cases. 

Would be good to add tests for these as well.

From a numerical point of view it might be useful to optimize the formula for small success or fail probabilities (or possibly rather the ratios in the formula). I imagine eg that it could be an advantage (if possible, just speculating) to use `log(x)` and `log1p(-x)` if `x` is close to 1 instead of `log(x)` and `log(z) = log(1 - x)` (IIRC `failprob` is just computed as `1 - succprob`).

Reply:
The constructor throws for zero or one parameter, but I will consider what happens for small x, y, z, or w.

Reply:
I'm not quite sure what to do here. Could you suggest a few test cases?

Reply:
Initially, I thought one should test `x = 0`, `z = 0`, and possibly some other zero values and their combinations. But I missed that these values can't occur (at least not if one calls the outer constructor), I guess it would be mostly interesting to see if something like `p = Geometric(nextfloat(0.0))` and `Geometric(prevfloat(1.0))` works correctly or if one should reformulate everything without using `failprob` which always involves a computation of `1-x`.